packages/xo-server/src/xapi/index.js

```diff
@@ -658,23 +658,32 @@ export default class Xapi extends XapiBase {
     // ensure the vm record is up-to-date
     vm = await this.barrier($ref)
 
+    if (!force && 'destroy' in vm.blocked_operations) {
+      throw forbiddenOperation('destroy', vm.blocked_operations.reason)
+    }
+
+    if (
+      !forceDeleteDefaultTemplate &&
+      vm.other_config.default_template === 'true'
+    ) {
+      throw forbiddenOperation('destroy', 'VM is default template')
+    }
```

Comment:
I moved the checks before to avoid any side effects when the deletion will fail.